### PR TITLE
feat: add protobuf compatibility checks

### PR DESCRIPTION
**Pull request by harikrishnakanchi**
- [x] add compatibility checks
- [x] Integrate to main flow
- [x] Protobuf, Avro, JSON schema validation
- [x] Avro schema compatibility checks

### CODE REVIEW

File: server/protobuf/compatibility.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+		compareFields(currentField, prevField, diffs)
+	}
+}
+
+func compareFields(currentField, prevField protoreflect.FieldDescriptor, diffs *compatibilityErr) {
+	if prevField.JSONName() != currentField.JSONName() {
+		diffs.add(fieldNameChange, prevField, `JSON field name changed from "%s" to "%s"`, prevField.JSONName(), currentField.JSONName())
+	}
+	name := prevField.Name()
+	if prevField.Cardinality().IsValid() && prevField.Cardinality().String() != currentField.Cardinality().String() {
```
[the remark this answers is not in the document]
Comment (harikrishnakanchi) [reply server/protobuf/compatibility.go:152]: ya. added label change test now.

File: server/protobuf/compatibility.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+	fieldTypeChange
+	enumDelete
+	enumValueDelete
+	enumValueDeleteWithoutReservedNumber
+	enumValueDeleteWithoutReservedName
+	enumValueNameChange
+	enumValueNumberChange
+)
+
+var (
```
[the remark this answers is not in the document]
Comment (pyadav) [reply server/protobuf/compatibility.go:30]: As per @harikrishnakanchi  We will do this later

[the remark this answers is not in the document]
Comment (harikrishnakanchi) [reply server/protobuf/compatibility.go:31]: Let's take compatibility for ONE_OF fields as a new feature request.

File: server/domain/schema.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+type SchemaFile struct {
+	ID     string
+	Types  []string
+	Fields []string
+	Data   []byte
+}
+
+// SchemaRepository for Schema
+type SchemaRepository interface {
```
[the remark this answers is not in the document]
Comment (harikrishnakanchi) [reply server/domain/schema.go:26]: We will tackle that in separate PR.

**Comment by pyadav** [comment]
LGTM

**Pull request merged**